helm/qontract-reconcile/templates/template.yaml

```diff
@@ -127,8 +127,102 @@ objects:
                 key: log_group_name
           command: ["/bin/sh", "-c"]
           args:
+          # generate /etc/vector/vector.yaml
+          # and generate /fluentd/etc/fluent.conf
           - |
-            # generate fluent.conf
+            ### APPSRE-8096: Vector Section Begins
+            mkdir -p /etc/vector
+            cat > /etc/vector/vector.yaml <<EOF
+            sources:
+              qr:
+                type: file
+                include:
+                  - /fluentd/log/integration.log
+                read_from: beginning
+                data_dir: /fluentd/log
+
+            transforms:
+              ### De-spaminate
+              despam_events:
+                type: filter
+                inputs:
+                  - qr
+                condition:
+                  type: datadog_search
+                  source: -"using gql endpoint" AND -"Certificate did not match expected hostname"
+
+              ### Craft a JSON document that has escaped double quotes and changes the event from the 
+              ### Vector-default `message` to `text`, which is require by Slack and Google Chat.
+              rename_message_webhooks:
+                type: remap
+                inputs:
+                  - despam_events
+                source: |-
+                  .message = join!(
+                  [
+                    "{\"text\":\"",
+                    format_timestamp!(
+                      now(),
+                      "%Y-%m-%d %H:%M:%S %z"
+                    ),
+                    " ",
```

Comment:
why the extra space here?

Reply:
Up on line 161 there is a `join!` command which is joining an array of strings into one single string. The `" ",` includes a literal "` `" into the array so that there's a space after the timestamp and before the `[PODNAMEHERE]` string.

Reply:
Gotcha, makes sense to me